src/coordinator/mod.rs

```diff
@@ -377,10 +378,11 @@ impl Coordinator {
 
                     Ok(ReadInput::ReadInput1(r1)) => entry.set_read_input_1(r1),
                     Ok(ReadInput::ReadInput2(r2)) => entry.set_read_input_2(r2),
-                    Ok(ReadInput::ReadInput3(r3)) => {
-                        let datalog = r3.datalog;
+                    Ok(ReadInput::ReadInput3(r3)) => entry.set_read_input_3(r3),
+                    Ok(ReadInput::ReadInput4(r4)) => {
+                        let datalog = r4.datalog;
 
-                        entry.set_read_input_3(r3);
+                        entry.set_read_input_4(r4);
```

Comment:
Actually I think I've realised this change might break older inverters that don't send this 4th packet. Now those inverters will never trigger the `all` packet :(

This is an old bodge coming home to bite me,  might have to have a think about how to fix this up.

Reply:
I'm actually having that issue currently with my 18kpv where the All packet isn't being sent so all in mqtt isn't updated.

Reply:
So as a temporary workaround for this I'm considering a config option to tell lxp-bridge when to send the `all` packet; probably just `publish_mqtt_all_after: 3` or something (then those who have a generator can set it to 4).

Longer term, I think I'd like to turn on `publish_individual_input` by default, then have HomeAssistant configuration watch for those instead, and ultimately retire the 1/2/3/4/all packets. They were a bit of an early poor design decision really.

Reply:
I was wondering how other inverters would handle the 4th as well.

@jgulick48 what version are you running on the 18k that doesn't have these? I'm on FAAB-1717 from January (not latest) - http://os.solarcloudsystem.com/#/releaseLog?deviceType=LXP_LB_8_12K

Reply:
I'm on FAAB-1818. They are there, but for some reason the all packet doesn't get sent to mqtt unless I manually trigger a read input 3.

Reply:
#249 underway with a new approach to packet decoding which should help with more than a few outstanding issues.